packages/autorest.python/autorest/preprocess/__init__.py

```diff
@@ -307,19 +320,36 @@ def update_client(self, yaml_data: Dict[str, Any]) -> None:
         yaml_data["builderPadName"] = to_snake_case(prop_name)
         for og in yaml_data["operationGroups"]:
             for o in og["operations"]:
-                for p in o["parameters"]:
+                idx_if_match = -1
```

Comment:
i feel there has to be a cleaner way of writing this code, i'm having a hard time following what it's supposed to do